plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/AdvancedConnectionEditor.java

```diff
@@ -351,11 +351,19 @@ public boolean isViewFor(Object object) {
 	}
 
 	public Map<String, Object> getExtendedProperties() {
+		if( map != null ) {
```

Comment:
why do we want to store an instance variable aka a shared additional state that we need to sync?